narwhals/_duckdb/dataframe.py

```diff
@@ -212,23 +214,40 @@ def filter(self, predicate: DuckDBExpr) -> Self:
 
     @property
     def schema(self) -> dict[str, DType]:
-        if self._cached_schema is None:
+        if self._cached_native_schema is None:
             # Note: prefer `self._cached_schema` over `functools.cached_property`
             # due to Python3.13 failures.
-            self._cached_schema = {
-                column_name: native_to_narwhals_dtype(duckdb_dtype, self._version)
-                for column_name, duckdb_dtype in zip(
-                    self.native.columns, self.native.types
+            self._cached_native_schema = dict(zip(self.columns, self.native.types))
```

Comment:
rather than caching the Narwhals schema (which may change, if the user modifies the connection time zone! !!!), we cache the native schema (which is the more potentially expensive one to calculate)